### ISSUE TITLE
add_smtp_user setting is not applied to ARC-Authentication-Results header

### ISSUE THREAD

**Issue by cygery** [rspamd/rspamd#2073]
### Classification (Please choose *one* option):

* [ ] Crash/Hang/Data loss
* [ ] WebUI/Usability
* [ ] Serious bug
* [ ] Other bug
* [X] Feature
* [ ] Enhancement

(Unsure if bug or feature.)

### Reproducibility (Please choose *one* option):

* [X] Always
* [ ] Sometimes
* [ ] Rarely
* [ ] Unable
* [ ] I didn’t try
* [ ] Not applicable

### Rspamd version:

1.7.0

### Operation system, CPU, memory and environment:

any

### Description (Please provide a descriptive summary of the issue):

Setting the `add_smtp_user` setting to `false` works fine to hide the `smtp.auth` field in the `Authentication-Results` header but it doesn't affect the `ARC-Authentication-Results` header.

### Compile errors (if any):

N/A

### Steps to reproduce:

- Set `add_smtp_user` to `false`
- Send an email

### Expected results:

The `ARC-Authentication-Results` header does not contain an `smtp.auth` field.

### Actual results:

The `ARC-Authentication-Results` header contains an `smtp.auth` field.

### Debugging information (see details [here](https://rspamd.com/doc/faq.html#how-to-figure-out-why-rspamd-process-crashed)):

N/A

### Configuration:

`/etc/rspamd/local.d/milter_headers.conf`:
```
routines {
    authentication-results {
        add_smtp_user = false;
    }
}
```

### Additional information:

I'm not sure what the expected behavior is. `arc.lua` creates the `ARC-Authentication-Results` header using `auth_results.gen_auth_results(task)` without allowing for user configuration like `milter_headers.lua` does. I'd like to be able to hide the `smtp.auth` field in the `ARC-Authentication-Results` header, using either the same config as for the `Authentication-Results` header (see above) or using an additional config block (maybe in `arc.conf`). My current workaround is to patch `lua_auth_results.lua` to set the default of `add_smtp_user` to `false`.

**Comment by hjelmua** [comment]
Setting add_smtp_user to false in lua_auth_results.lua also removes the duplicate of "sign_headers" that was both in DKIM and ARC. 

"h=from:sender:reply-to:subject:date:message-id:to:cc:mime-version:content-type:content-transfer-encoding:resent-to:resent-cc:resent-from:resent-sender:resent-message-id:in-reply-to:references:list-id:list-owner:list-unsubscribe:list-subscribe:list-post;"

**Comment by hjelmua** [comment]
There is a simpler way to fix the issue. Many guides have #allow_username_mismatch = true; in local.d/arc.conf. If one only uses 
path = "/var/lib/rspamd/dkim/$selector.key";
selector = "THEKEY";
in the local.d/arc.conf it works fine.

and does not have skip_local = false in milter_headers.conf it works fine.

But since most guides online have the values a future version of add_smtp_user = false; should remove the smtp.auth field in ARC-Auth as well.

**Comment by bjoe2k4** [comment]
I think there should be a single file of configuration for both `ARC-Authentication-Results` as well as `Authentication-Results`, after all one should merely be the copy of the other, with an `i=` property prepended. Right now `Authentication-Results` can be modified in `milter_headers.conf`, while `ARC-Authentication-Results` is hardcoded in `lua_auth_results.lua`, as far as i can tell. If @vstakhov can advise whether such a central configuration is desired and where it should be located - `milter_headers.conf` or a new config file - i'd like to have a look at it and come up with a pull request.

**Comment by bjoe2k4** [comment]
bump

**Comment by DrGeek** [comment]
bump 2 :D ?

**Comment by bjoe2k4** [comment]
bump 3! @vstakhov

**Comment by martbock** [comment]
Having the same problem and always patching `lua_auth_results.lua` is not a long-term solution.

**Comment by DrGeek** [comment]
the situation is still the same :(

**Comment by ColinFinck** [comment]
Seems to be fixed since 1.9.3
